cluster/stmt.go

```diff
@@ -107,8 +108,15 @@ func GetCallerProject() string {
 
 	// If not a snap build path, the project may be in a go module path of the form .../project@version....
 	before, _, ok := strings.Cut(file, "@")
-	if ok && filepath.Base(before) != "" {
-		return filepath.Base(before)
+	base := filepath.Base(before)
```

Comment:
I am trying to come up with a scenario where you would actually have different callers/paths in the same program?
Wouldn't the return value of `GetCallerProject()` always be identical during runtime? 

If that is the case this func feels overly complex given we could just return a string which isn't `microcluster` for the `stmtsByProject` map.

Reply:
Or is it required for the case of MicroCloud where we import the Micro* projects and each of them calls microcluster independently? 

Why not just choosing a UUID during startup of the microcluster daemon and using this instead?

Reply:
I think moving away from this is a potential idea for v3 but I don't want to worry about unintended consequences of moving in a different direction for the LTS.

If I remember right, the goal of this is to be able to distinguish which registered statement came from which project. So we need to still parse the caller function of `RegisterStmt` to determine which package it belongs to. It may be called from microcluster, from a package importing microcluster (like MicroCloud), or via another microcluster project (like MicroCloud importing MicroOVN). 

The returned identifier doesn't have to be the name but I'm not sure if moving to a UUID would necessarily reduce the complexity, since we would still need to parse the project name out to determine if we need to re-use a UUID or generate a new one.

Reply:
I should mention, in the latter case (MicroCloud importing MicroOVN), we actually don't want to prepare those statements because there will never be a use-case for them in MicroCloud, since we can't directly access the database of MicroOVN. 

So part of this helper is ensuring we only import from the current project and microcluster, and ignore all other `RegisterStmt` calls in the import tree.

Reply:
I was thinking this could be an additional field of the `type Daemon struct`.

But yes `RegisterStmt` seems problematic here as we cannot easily access the daemon and it stores the stmt's in a package level var.

Then I guess it's ok for now. Looks not to be an easy fix we could just slide into the LTS. But could you please log an issue so we can follow up on this?

Reply:
https://github.com/canonical/microcluster/issues/220